Fix issue #1769

fucking-signup opened the pull request:
## Description
This PR:
- Fixes issue where `like_num` returned `False` when numeric word was not lowercased (#1769).
- Adds checks for `like_num` in ordinal words lists, where ordinal words were specified but left unused (French, Dutch and Portuguese). Previously, only Danish checked for `like_num` in ordinals words list.
- Updates code example in [Adding Languages](https://spacy.io/usage/adding-languages) in Usage section on the website according to proposed changes.

### Types of change
Bugfix and enhancement

## Checklist
- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

fucking-signup commented:
Hmm. The build has failed on AppVeyor but it is not related to proposed changes. Is there a way to re-run the build process?

ines commented:
@fucking-signup Thanks a lot for the detailed PR! 🙏  Just restarted the Appveyor build... not sure what went wrong there – sometimes the builds can be randomly flakey.

fucking-signup commented:
@ines great, thank you! 😊

fucking-signup commented:
Interesting. Still fails on Windows under Python 2.7, yet no errors related to changes.

ines commented:
Yeah, this is strange... I also don't think it's related to some weird recursion or other issue caused by the language data imports, because that should fail more consistently? 

I might just try it again tomorrow, because who knows... if it still fails, we'll try and rebuild master, and see if we get the same error there...

fucking-signup commented:
Could be an issue with AppVeyor's environment with Python 2.7?
There are no issues with Python 3.6 on any platform and no issues with Python 2.7 on Linux.

honnibal commented:
One thing that can go wrong on Windows Python2.7 but not other environments is that `Path.open` won't necessarily default to a utf8 encoding on that platform, while on the other environments it does. If we have any calls to `Path.open()` without the `encoding` arg explicit, this can cause an error.

However, I can't match that problem up to this error --- I would've expected an encoding bug to trigger an error right away.

I wonder whether we're just on the edge of the Appveyor VM memory limit, and indexing a few extra tests with pytest is just enough to tip things over? This could be exacerbated by the double parametrize, which maybe pytest handles inefficiently?

fucking-signup commented:
Sure, will try today!

fucking-signup commented:
@honnibal yup, we are on edge on Appveyor VM. Build on Windows under Python 2.7 just succeeded after I removed all tests but one.

ines commented:
Thanks for the fix, merging now! 👍 

Sorry this was kinda frustrating – but it's still a good thing the memory limit came up. I guess we'll have to tidy up the tests soon!

fucking-signup commented:
@ines wait, I didn't fix it haha
Just verified that this was an issue. Fixing now.

ines commented:
Ahhh, sorry, should have checked this more carefully – I thought you had just simplified the tests. I don't think we need to revert everything in #1808 – I think it'd be fine to just update the tests. I'm pretty confident it'll work as expected, considering the fix is fairly minor.

I think one of the main issues might indeed be the double parametrize – if I remember correctly, this will run the test for all combinations of the available parameters, which is pretty overkill in our case. So we could simplify it to something like this:

```python
@pytest.mark.parametrize('word', ['onze', 'quadragésimo')])
def test_pt_lex_attrs(word):
    assert pt_like_num(word) == pt_like_num(word.upper())
```

What do you think?

fucking-signup commented:
@ines no problem, I should have expressed myself more clearly :)
I have rewritten tests, removed parametrize altogether and used fixtures as suggested by Matthew. Still didn't work.

fucking-signup commented:
Just checked your suggestion, fails as well 😞

ines commented:
Thanks for trying! I think I might know what's going on then: when importing a sub-mobule from the language data, the top-level module is imported as well. The language data can get pretty large (especially with all the lemmatization data etc.), which is why the respective class is usually lazy-loaded only when it's needed.

Here, the tests are importing 7 languages in total, including some pretty large ones. So what we could do instead is lazy-load them within the tests instead (i.e. via `util.get_lang_class()`), mark the tests with `@pytest.mark.slow` and only leave one (e.g. the English one) as a sanity check that runs on each build.

fucking-signup commented:
@ines great idea!
Will do that.

fucking-signup commented:
It works! Thank you! 😊 
Created new PR.
